std/math/fp2.asm

```diff
@@ -20,9 +20,8 @@ use std::prover::eval;
 /// f = x^2 - 7
 /// f.is_irreducible()
 
-/// An element of the extension field over the implied base field (which has to be either
-/// the Goldilocks or the BN254 field) relative to the irreducible polynomial X^2 - 7,
-/// (This irreducible polynomial also works for Mersenne31)
+/// An element of the extension field over the implied base field (which has to be one
+/// of the field elements: Goldilocks, BN254, BabyBear, M31) relative to the irreducible polynomial X^2 - 11,
```

Comment:
Do we have a test somewhere that verifies that it is really irreducible on all the fields we care about?

Reply:
We should somehow check that `Fp2` can only be constructed for the known fields. But that would mean that this check has to be run every time we call some function here...

Reply:
Maybe the method returning the irreducible polynomial should be available in the field's trait, so it could be made to work with any field.

Reply:
The problem is that in pil, the field is not really a type. We could use the `known_field` function, but then we would have to do it at every multiplication.

Reply:
My idea was to introduce a `get_f2_irreducible()` function, but yeah, it would have to be called everywhere it is needed.

Reply:
We have the sage test in here as a comment: https://github.com/powdr-labs/powdr/blob/b2a06c2b03997f77b04ae2b0e41f9bcd996f83e3/std/math/fp2.asm#L13-L21 (x^2 - 7 is wrong, should be x^2 - 11). I would like to add the code inside the following gist as a comment: https://gist.github.com/onurinanc/ccb37ae5dde012dc8bb67617a0f89376

Reply:
If we had a constructor for Fp2 we could do the check when creating a new fp2 instance, but we are using Fp2::Fp2()

Reply:
Well, this PR doesn't solve the problem, but it hasn't created it, either, so I am approving it.